### DISCUSSION TITLE
Caching for individual build artifacts from an aggregate build script?

### DISCUSSION THREAD

**Discussion by ivancuric** [vercel/turborepo#1536] [Help] [answered]
I'm working on a monorepo containing webassembly. The build script actually runs multiple separate scripts via `npm-run-all`:
1) configure the cmake build (`configure-build`)
2) run cmake to compile the emscripten wasm artifacts (`build:wasm`)
3) run vite to compile JS (`build:js`)
4) run `tsc` and `api-extractor` to generate the types (`build:types`)

So in the end I have:
```json
"build": "npm-run-all -s configure-build build:wasm build:js build:types"
```

Is there a way to separately cache the outputs of each task/script, but that it works in the aggregated build task? Like, 99% of the time I just want the JS to recompile, but the C++ code should only recompile if the cmake build or cpp sources change.

Importantly, all 3 steps are in the same package in the monorepo.

So technically if I were to separate these scripts it would look something like:
```js
    "configure-build": {
      "dependsOn": ["^build"],
      "outputs": ["build/**"],
      "inputs": ["**/CMakeLists.txt", "**/conanfile.txt"]
    },
    "build:wasm": {
      "dependsOn": ["^build"],
      "outputs": ["resources/**"],
      "inputs": ["build/**", "binding/**"]
    },
    "build:types": {
      "dependsOn": ["^build"],
      "outputs": ["types/**"],
      "inputs": ["src/**"]
    },
    "build:vite": {
      "dependsOn": ["^build"],
      "outputs": ["dist/**"],
      "inputs": ["src/**"]
    }
```

However, this won't work as there is no core `build` script

**Comment by anthonyshew** [comment, marked as answer]
Using `npm-run-all` with `turbo` is a bit of an anti-pattern. You're using two task runners so they may step on each other.

What you've described can be achieved through a reworked `turbo.json` file, output capturing and parallelization included.
